### ISSUE TITLE
Error generating accessibility requirements: no available topology found

### ISSUE THREAD

**Issue by 0xfourzerofour** [linode/linode-blockstorage-csi-driver#433]
I am getting the following error when running both linode-ccm and the csi driver

```

E0601 00:32:47.814395       1 controller.go:974] "Unhandled Error" err="error syncing claim \"bacd0750-5c8b-4268bc15-00b96c631bdb\": failed to provision volume with Storage Class \"linode-block-storage-retain\": error generating accessibility requirements: no available topology found" logger="UnhandledError" 

```

```
ubectl get csidrivers
NAME                      ATTACHREQUIRED   PODINFOONMOUNT   STORAGECAPACITY   TOKENREQUESTS   REQUIRESREPUBLISH   MODES        AGE
linodebs.csi.linode.com   true             true             false             <unset>         false               Persistent   95m
```

```
kubectl get csinode worker-0 -o yaml
apiVersion: storage.k8s.io/v1
kind: CSINode
metadata:
  annotations:
    storage.alpha.kubernetes.io/migrated-plugins: kubernetes.io/aws-ebs,kubernetes.io/azure-disk,kubernetes.io/azure-file,kubernetes.io/cinder,kubernetes.io/gce-pd,kubernetes.io/portworx-volume,kubernetes.io/vsphere-volume
  creationTimestamp: "2025-05-03T21:13:28Z"
  name: worker-0
  ownerReferences:
  - apiVersion: v1
    kind: Node
    name: worker-0
    uid: 9db67bfa-e22d-48e1-b031-17f8355d1392
  resourceVersion: "364"
  uid: 67ade405-f9a9-4ec1-9316-77b6444e8aae
spec:
  drivers: null

```

The driver is not registering to the cluster so the linode provisioner cannot run correctly.

Kube version: `1.32.4`
Kube framework: k0s

Here is what my helm config looks like to set up the storage driver + linode ccm

```
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: linode-ccm
  namespace: linode-system
spec:
  interval: 1m
  timeout: 5m 
  chart:
    spec:
      chart: ccm-linode
      version: "0.6.1"  
      sourceRef:
        kind: HelmRepository
        name: linode-ccm
        namespace: linode-system
      interval: 1m
  valuesFrom:
    - kind: Secret
      name: linode-api-token
      valuesKey: token
      targetPath: apiToken
  values:
    region: "us-east" 
    namespace: linode-system
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
              - key: node-role.kubernetes.io/control-plane
                operator: DoesNotExist
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  name: linode-block-storage
  namespace: linode-system
spec:
  interval: 1m
  timeout: 5m 
  chart:
    spec:
      chart: linode-blockstorage-csi-driver
      version: "1.0.5"  
      sourceRef:
        kind: HelmRepository
        name: linode-block-storage
        namespace: linode-system
      interval: 1m
  valuesFrom:
    - kind: Secret
      name: linode-api-token
      valuesKey: token
      targetPath: apiToken
  values:
    region: "us-east" 
    namespace: linode-system

```

Any help would be great!

**Comment by 0xfourzerofour** [comment]
Note I need to set the affinity to controller nodes because k0s has clear controller/worker separation
